src/ui/window.cpp

```diff
@@ -0,0 +1,182 @@
+#include <glad/glad.h>
+
+// GLAD must be included first. clang-format sorts includes, hence this comment
+// here.
+
+#include "window.hpp"
+#include "movable_rect.hpp"
+
+
+static GLFWwindow* window = nullptr;        // Window instance
+
+// List of movable rectangles. This will be important in saving config.
+static List<UI_Movable_Rect_S*> movable_rectangles;
+static UI_Movable_Rect_S* currently_moved_rectangle = nullptr;  // Currently dragged rectangle.
+static f32 last_x, last_y;    // Last mouse x and y position
```

Comment:
this should be abstracted to a system. prefer to not use static data

Reply:
"cursor_position_callbacks" and "mouse_button_callbacks" are functions for which these variables are needed and it is not possible to pass them as a parameter because they must follow the pattern imposed by GLFW. 
I don't know if there is any other way to pass these values ​​other than through static variables

Reply:
create an input system that will be updated by the glfw callback functions. this will allow us to expose input to other systems, e.g. ui